ssl/statem/statem_lib.c

```diff
@@ -1039,14 +1039,25 @@ WORK_STATE tls_finish_handshake(SSL *s, WORK_STATE wst, int clearbufs, int stop)
     int cleanuphand = s->statem.cleanuphand;
 
     if (clearbufs) {
-        if (!SSL_IS_DTLS(s)) {
+        if (!SSL_IS_DTLS(s)
+#ifndef OPENSSL_NO_SCTP
+                /*
+		 * RFC6083: SCTP provides a reliable and in-sequence transport service for DTLS
+                 * messages that require it. Therefore, DTLS procedures for retransmissions
+		 * MUST NOT be used.
+                 * Hence the init_buf can be cleared when DTLS over SCTP as transport is used.
+                 */
+	        || BIO_dgram_is_sctp(SSL_get_wbio(s))
+#endif
+	        ) {
             /*
-             * We don't do this in DTLS because we may still need the init_buf
+             * We don't do this in DTLS over UDP because we may still need the init_buf
```

Comment:
Just a curious question maybe: Why did you add "over UDP"? I thought Datagram TLS was synonymous to TLS over UDP? Is this just a hendiadys, or are there other DTLS scenarios?

Reply:
Oh, forget my stupid question. I should have googled SCTP first ;-)